code/modules/projectiles/boxes_magazines/_box_magazine.dm

```diff
@@ -131,4 +131,10 @@
 	var/turf_mag = get_turf(src)
 	for(var/obj/item/ammo in stored_ammo)
 		ammo.forceMove(turf_mag)
-		stored_ammo -= ammo
\ No newline at end of file
+		stored_ammo -= ammo
+
+/obj/item/ammo_box/magazine/handle_atom_del(atom/A)
+	if(A.type == ammo_type)
+		for(var/obj/item/ammo_casing/AC in stored_ammo)

```

Comment:
you can just use `stored_ammo.Cut()` to empty the list. But why would you delete all the ammo in the gun just because one was deleted?

Reply:
yeah that's one problem that I saw. I mean I don't think most people make magazines of guns with different bullet types but I can do that.